File: benchmarks/cross/cases/storage/scope.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func oneScope(path string) (count int, err error) {
	file, err := os.Open(path)
	if err != nil {
		return 0, err
	}
	defer func() {
		closeErr := file.Close()
		if err == nil {
			err = closeErr
		}
	}()
	var octet [1]byte
	count, err = file.Read(octet[:])
	return count, err
}

func main() {
	if len(os.Args) != 3 {
		fmt.Fprintln(os.Stderr, "usage: scope SCOPES FIXTURE")
		os.Exit(2)
	}
	scopes, err := strconv.Atoi(os.Args[1])
	if err != nil || scopes <= 0 {
		fmt.Fprintln(os.Stderr, "scope count must be positive")
		os.Exit(2)
	}
	checksum := 0
	for i := 0; i < scopes; i++ {
		count, err := oneScope(os.Args[2])
		if err != nil {
			fmt.Fprintln(os.Stderr, err)
			os.Exit(1)
		}
		checksum += count
	}
	fmt.Printf("scopes=%d bytes=%d checksum=%d\n", scopes, checksum, checksum)
}

File: benchmarks/cross/cases/storage/scope.cpp
#include <fstream>
#include <iostream>
#include <stdexcept>
#include <string>

int one_scope(const std::string& path) {
  std::ifstream source(path, std::ios::binary);
  if (!source) {
    throw std::runtime_error("cannot open fixture: " + path);
  }
  char octet = 0;
  source.read(&octet, 1);
  if (source.gcount() != 1) {
    throw std::runtime_error("fixture did not contain one byte");
  }
  return 1;
}

int main(int argc, char** argv) {
  if (argc != 3) {
    std::cerr << "usage: scope SCOPES FIXTURE\n";
    return 2;
  }
  try {
    const int scopes = std::stoi(argv[1]);
    if (scopes <= 0) {
      throw std::runtime_error("scope count must be positive");
    }
    int checksum = 0;
    for (int i = 0; i < scopes; ++i) {
      checksum += one_scope(argv[2]);
    }
    std::cout << "scopes=" << scopes
              << " bytes=" << checksum
              << " checksum=" << checksum << '\n';
  } catch (const std::exception& error) {
    std::cerr << error.what() << '\n';
    return 1;
  }
}
